### PR TITLE
feat: send test mail

### LINKED ISSUE #4504: SMTP send test mail
As a administrator I want to be able to configure my SMTP and send a test email to find out if everything works before activating it for all my customers.

**Acceptance criteria**
- [ ] I can create a SMTP configuration and put it in a preview mode
- [ ] I can send a test email from a SMTP no matter of the state (preview/active)
- [x] I can deactivate an SMTP
- [x] I can activate an SMTP
- [x] I am allowed to have no active SMTP configuration
- [x] I am able to see the error messages that happen when an email can't be sent
- [ ] I am able to reset my SMTP configuration to the default of ZITADEL

### PR DESCRIPTION
**Pull request by doncicuto**
# Which Problems Are Solved

- Zitadel doesn't have a way to test SMTP settings either before creating a new provider or once the SMTP provider has been created.
- Zitadel SMTP messages can be more informative for usual errors

# How the Problems Are Solved

- A new step is added to the new/update SMTP provider wizard that allows us to test a configuration. The result is shown in a text area.
- From the table of SMTP providers you can test your settings too.
- The email address to send the email is by default the email address for the logged in user as suggested.
- Some of the SMTP error messages have been changed to give more information about the possible situation. For example: could not contact with the SMTP server, check the port, firewall issues... instead of could not dial

Here's a video showing this new option in action:

https://github.com/zitadel/zitadel/assets/30386061/50128ba1-c9fa-4481-8eec-e79a3ca69bda

# Additional Changes

Replace this example text with a concise list of additional changes that this PR introduces, that are not directly solving the initial problem but are related.
For example:
- The docs explicitly describe that the property XY is mandatory
- Adds missing translations for validations.

# Additional Context

- Closes #4504

### CODE REVIEW

File: proto/zitadel/settings.proto [hunk 1]
```diff
@@ -95,6 +95,16 @@ message SMTPConfig {
   string id = 10;
 }
 
+message SMTPTestResult {
```
Comment (stebenz) [inline proto/zitadel/settings.proto:98]: Is this message used anywhere?

File: proto/zitadel/admin.proto [hunk 2]
```diff
@@ -4254,6 +4288,82 @@ message RemoveSMTPConfigResponse {
...
+        (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+            example: "\"197f0117-529e-443d-bf6c-0292dd9a02b7\"";
+        }
+    ];
+    string password = 6 [
+        (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+            example: "\"this-is-my-password\"";
+        }
+    ];
+    string test_address = 7 [
```
Comment (stebenz) [inline proto/zitadel/admin.proto:4347]: Not quite sure, should we leave it like this or something like reciever_address?

File: internal/api/grpc/admin/smtp.go [hunk 3]
```diff
@@ -128,3 +129,33 @@ func (s *Server) DeactivateSMTPConfig(ctx context.Context, req *admin_pb.Deactiv
 		Details: object.DomainToAddDetailsPb(result),
 	}, nil
 }
+
+func (s *Server) TestSMTPConfigById(ctx context.Context, req *admin_pb.TestSMTPConfigByIdRequest) (*admin_pb.TestSMTPConfigByIdResponse, error) {
+	instanceID := authz.GetInstance(ctx).InstanceID()
+	resourceOwner := instanceID // Will be replaced when orgs have smtp configs
+
+	err := s.command.TestSMTPConfigById(ctx, resourceOwner, req.Id, req.TestAddress)
```
Comment (stebenz) [inline internal/api/grpc/admin/smtp.go:137]: Can be combined with the 3 lines above, as it has to be changed anyway with logic changes.

File: internal/api/grpc/admin/smtp.go [hunk 4]
```diff
@@ -128,3 +129,33 @@ func (s *Server) DeactivateSMTPConfig(ctx context.Context, req *admin_pb.Deactiv
...
+	instanceID := authz.GetInstance(ctx).InstanceID()
+	config := smtp.Config{}
+	config.Tls = req.Tls
+	config.From = req.SenderAddress
+	config.FromName = req.SenderName
+	config.SMTP.Host = req.Host
+	config.SMTP.User = req.User
+	config.SMTP.Password = req.Password
+
+	err := s.command.TestSMTPConfig(ctx, instanceID, req.Id, req.TestAddress, &config)
```
Comment (stebenz) [inline internal/api/grpc/admin/smtp.go:155]: Same here with instanceID.

File: proto/zitadel/settings.proto (hunk 1 above)
Comment (doncicuto) [reply proto/zitadel/settings.proto:98]: Thanks @stebenz, fixing it

**Comment by doncicuto** [comment]
Hi @hifabienne, I think I've solved some issues so the review can be resumed

File: proto/zitadel/admin.proto [hunk 5]
```diff
@@ -503,6 +503,40 @@ service AdminService {
...
+        };
+
+        option (zitadel.v1.auth_option) = {
+            permission: "iam.write";
+        };
+
+        option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+            tags: "SMTP Provider";
+            summary: "Test SMTP Provider ";
+            description: "Test an SMTP provider identified by its is. After testing the provider, the users will receive information about the test results."
```
Comment (eliobischof) [inline proto/zitadel/admin.proto:519]:
```suggestion
            description: "Test an SMTP provider identified by its ID. After testing the provider, the users will receive information about the test results."
```

File: internal/static/i18n/en.yaml [hunk 6]
```diff
@@ -53,11 +53,24 @@ Errors:
...
     AlreadyDeactivated: SMS configuration already deactivated
+  SMTP:
+    NotEmailMessage: message is not EmailMessage
+    RequiredAttributes: subject, recipients and content must be set but some or all of them are empty
+    CouldNotSplit: could not split host and port for connect to smtp
+    CouldNotDial: could not contact with the SMTP server, check the port, firewall issues...
+    CouldNotDialTLS: could not contact with the SMTP server using TLS, check the port, firewall issues...
+    CouldNotCreateClient: could not create smtp client
+    CouldNotStartTLS: could not start tls
+    CouldNotAuth: could not add smtp auth, check if both your user and password are correct, if they're correct maybe your provider requires an auth method not supported by Zitadel
```
Comment (eliobischof) [inline internal/static/i18n/en.yaml:64]:
```suggestion
    CouldNotAuth: could not add smtp auth, check if both your user and password are correct, if they're correct maybe your provider requires an auth method not supported by ZITADEL
```

Can you correct this in the other files too, please?

File: internal/static/i18n/de.yaml [hunk 7]
```diff
@@ -53,11 +53,24 @@ Errors:
...
     AlreadyDeactivated: SMS Konfiguration ist bereits deaktiviert
+  SMTP:
+    NotEmailMessage: Die Nachricht ist nicht EmailMessage
+    RequiredAttributes: Betreff, Empfänger und Inhalt müssen festgelegt werden, aber einige oder alle davon sind leer
+    CouldNotSplit: Host und Port konnten für die Verbindung zu SMTP nicht getrennt werden
+    CouldNotDial: Es konnte keine Verbindung zum SMTP-Server hergestellt werden, Port überprüfen, Firewall-Probleme ...
+    CouldNotDialTLS: Es konnte keine Verbindung zum SMTP-Server über TLS hergestellt werden, überprüfen Sie den Port, Firewall-Probleme ...
+    CouldNotCreateClient: Der SMTP-Client konnte nicht erstellt werden
+    CouldNotStartTLS: TLS konnte nicht gestartet werden
+    CouldNotAuth: Die SMTP-Authentifizierung konnte nicht hinzugefügt werden. Überprüfen Sie, ob sowohl Ihr Benutzername als auch Ihr Passwort korrekt sind. Wenn sie korrekt sind, erfordert Ihr Anbieter möglicherweise eine Authentifizierungsmethode, die von Zitadel nicht unterstützt wird
```
Comment (eliobischof) [inline internal/static/i18n/de.yaml:64]:
```suggestion
    CouldNotAuth: Die SMTP-Authentifizierung konnte nicht hinzugefügt werden. Überprüfen Sie, ob sowohl Ihr Benutzername als auch Ihr Passwort korrekt sind. Wenn sie korrekt sind, erfordert Ihr Anbieter möglicherweise eine Authentifizierungsmethode, die von ZITADEL nicht unterstützt wird
```

File: console/src/app/modules/smtp-provider/smtp-provider.component.ts [hunk 8]
```diff
@@ -281,6 +300,33 @@ export class SMTPProviderComponent {
...
+    req.setTls(this.tls?.value ?? false);
+    req.setId(this.id ?? '');
+    req.setReceiverAddress(this.email ?? '');
+
+    this.service
+      .testSMTPConfig(req)
+      .then(() => {
+        this.resultClass = 'test-success';
+        this.isLoading.set(false);
+        this.testResult = 'Your email was succesfully sent';
```
Comment (eliobischof) [inline console/src/app/modules/smtp-provider/smtp-provider.component.ts:321]: Can we translate this?

**Review by eliobischof — CHANGES_REQUESTED** [review]
Would you mind also adding the new Swedish translations?

**Comment by doncicuto** [comment]
Thanks @eliobischof I'll prepare the changes right away

**Comment by doncicuto** [comment]
Hi @eliobischof I've updated the PR with your suggestions including Swedish translation. Thanks for your review!

**Review by eliobischof — APPROVED** [review]

**Pull request merged**